core/rend/vulkan/oit/oit_pipeline.h

```diff
@@ -162,45 +162,43 @@ class OITDescriptorSets
 			perPolyDescSet = perPolyAlloc.alloc();
 			std::vector<vk::WriteDescriptorSet> writeDescriptorSets;
 
-			vk::DescriptorImageInfo imageInfo0;
 			if (poly.texture != nullptr)
 			{
-				imageInfo0 = vk::DescriptorImageInfo{ samplerManager->GetSampler(poly.tsp), ((Texture *)poly.texture)->GetReadOnlyImageView(),
-						vk::ImageLayout::eShaderReadOnlyOptimal };
-				writeDescriptorSets.emplace_back(perPolyDescSet, 0, 0, 1, vk::DescriptorType::eCombinedImageSampler, &imageInfo0, nullptr, nullptr);
+				vk::DescriptorImageInfo imageInfo0(samplerManager->GetSampler(poly.tsp), ((Texture *)poly.texture)->GetReadOnlyImageView(),
+						vk::ImageLayout::eShaderReadOnlyOptimal);
+				writeDescriptorSets.emplace_back(perPolyDescSet, 0, 0, vk::DescriptorType::eCombinedImageSampler, imageInfo0);
 			}
-			vk::DescriptorImageInfo imageInfo1;
+
 			if (poly.texture1 != nullptr)
 			{
-				imageInfo1 = vk::DescriptorImageInfo{ samplerManager->GetSampler(poly.tsp1), ((Texture *)poly.texture1)->GetReadOnlyImageView(),
-					vk::ImageLayout::eShaderReadOnlyOptimal };
-				writeDescriptorSets.emplace_back(perPolyDescSet, 1, 0, 1, vk::DescriptorType::eCombinedImageSampler, &imageInfo1, nullptr, nullptr);
+				vk::DescriptorImageInfo imageInfo1(samplerManager->GetSampler(poly.tsp1), ((Texture *)poly.texture1)->GetReadOnlyImageView(),
+					vk::ImageLayout::eShaderReadOnlyOptimal);
+				writeDescriptorSets.emplace_back(perPolyDescSet, 1, 0, vk::DescriptorType::eCombinedImageSampler, imageInfo1);
```

Comment:
imageInfo1 is used be reference but goes out of scope

Reply:
Fixed